src/index.js

```diff
@@ -1,21 +1,32 @@
 import visit from 'unist-util-visit';
 import qs from 'query-string';
 
-module.exports = function gatsbyRemarkCodeTitles({ markdownAST }, { className: customClassName }) {
+module.exports = function gatsbyRemarkCodeTitles(
+  { markdownAST },
+  { className: customClassName }
+) {
   visit(markdownAST, 'code', (node, index) => {
-    const [language, query] = (node.lang || '').split(':');
-    const { title } = qs.parse(query);
+    const [language, ...params] = (node.lang || '').split(':');
+    const query = params.join('&');
+    const options = qs.parse(query);
+    const title = options.title;
```

Comment:
What about:

```suggestion
    const { title, ...rest } = options;
```

and then you can use ...rest later to re-construct?